src/Field.php

```diff
@@ -489,7 +489,12 @@ public function setNull(): self
      */
     public function compare($value): bool
     {
-        return $this->get() == $value;
+        if ($this->owner->persistence === null) {
+            return (string) $this->normalize($this->get()) === (string) $this->normalize($value);
```

Comment:
Why the explicit (string) typecast? Shouldn't normalize() return same data type if $value equals $this->get() anyway?

Reply:
normalize does not cast if type is not defined, might be worth later to update...